ldap/src/main/java/org/springframework/security/ldap/authentication/ad/ActiveDirectoryLdapAuthenticationProvider.java

```diff
@@ -397,6 +404,15 @@ public void setContextEnvironmentProperties(Map<String, Object> environment) {
 		this.contextEnvironmentProperties = new Hashtable<>(environment);
 	}
 
+	private void setAuthoritiesPopulator(LdapAuthoritiesPopulator authoritiesPopulator) {
+		Assert.notNull(authoritiesPopulator, "An LdapAuthoritiesPopulator must be supplied");
+		this.authoritiesPopulator = authoritiesPopulator;
+	}
+
+	public LdapAuthoritiesPopulator getAuthoritiesPopulator() {
```

Comment:
Please remove the getter, we can always add it later when the need arises.

Reply:
Usually, I try and wait for that request. Historically, Spring Security has added very few getters.

Most often, an application knows the value anyway because it invoked the setter at configuration time. Strategically, Spring Security tries to keep its components write-only to afford us more internal flexibility to adapt the instance without needing to find a way to reconstitute it when the getter is called.

Reply:
Thanks for a clarification, appreciate it. Done, removed the getter.